### PR TITLE
Avoid redefining builting id function

### PR DESCRIPTION
**Pull request by iblancasa**
Previously, the [builtin function id()](https://docs.python.org/2/library/functions.html#id) was redefined.

Thank you!

### CODE REVIEW

File: elasticsearch/client/ingest.py [hunk 1]
```diff
@@ -46,16 +46,16 @@ def delete_pipeline(self, id, params=None):
             'pipeline', id), params=params)
 
     @query_params('verbose')
-    def simulate(self, body, id=None, params=None):
+    def simulate(self, body, pipelineID=None, params=None):
```
Comment (georgepsarakis) [inline elasticsearch/client/ingest.py]: Usually in Python in order not to shadow builtins you can append an underscore: `id_`.
If you want to use this name, I think you should change it to `pipeline_id`.

File: elasticsearch/client/ingest.py [hunk 2]
```diff
@@ -46,16 +46,16 @@ def delete_pipeline(self, id, params=None):
             'pipeline', id), params=params)
```
Comment (georgepsarakis) [inline elasticsearch/client/ingest.py]: I guess the same should change should be applied in this method as well?

Comment (iblancasa) [reply elasticsearch/client/ingest.py]: Oh! Yes. I will change this ASAP. Thanks for your feedback!

**Comment by HonzaKral** [comment]
Thank you for the patch, @iblancasa, but I don't believe this change is worth breaking the API for some people. The `id` function is not being redefined except inside the API method where it is not needed so no actual harm is done.

In the beginning, when designing this library, we decided to rename the variable `type` to `doc_type` and leave the `id` as is. The motivation for this was that we felt that `type` is more widely used and people would adopt it as variable name in their code, possibly leading to some issues. With `id` this is less of a problem and also widely accepted throughout python codebases.

I know this is not ideal, but right now making the change would introduce more harm and confusion. Thank you for understanding!

**Pull request closed without merging**